curvefs/src/tools/curvefs_tool.h

```diff
@@ -196,32 +207,61 @@ class CurvefsToolRpc : public CurvefsTool {
      * as long as one succeeds, it returns true and ends sending
      */
     virtual bool SendRequestToServices() {
-        uint32_t failHostNumner = 0;
+        uint32_t failHostNumber = 0;
+        bool ret = false;
         for (const std::string& host : hostsAddr_) {
             SetController();
-            if (channel_->Init(host.c_str(), nullptr) != 0) {
-                std::cerr << "fail init channel to host: " << host << std::endl;
+            std::shared_ptr<brpc::ChannelOptions> channelOpt = nullptr;
+            if (isStreaming_) {
+                // set stream rpc client
+                channelOpt = std::make_shared<brpc::ChannelOptions>();
+                channelOpt->connection_group = "streaming";
+                StreamOptions streamOpt(FLAGS_rpcStreamIdleTimeoutMs);
+                connection_ = streamClient_->Connect(
+                    controller_.get(), receiveCallback_, streamOpt);
+                if (nullptr == connection_ || nullptr == receiveCallback_) {
```

Comment:
should you check `receiveCallback_ == nullptr` before Connect or just after `if (isStreaming_)`?